Classes/System/Solr/Node.php

```diff
@@ -129,28 +118,16 @@ public function getPassword(): string
         return (string)$this->getOption('password');
     }
 
-    /**
-     * @return int
-     */
-    public function getTimeout(): int
-    {
-        return $this->timeout;
-    }
-
     /**
      * Returns the path including api path.
      *
      * @return string
      */
-    public function getCoreBasePath()
+    public function getCoreBasePath(): string
     {
-        $api = ($this->apiVersion === 1) ? 'solr' : 'api';
-        $elements = [
-            $this->path,
-            $api
-        ];
-
-        return '/' . implode('/', $elements) . '/';
+        $pathWithoutLeadingAndTrailingSlashes = trim(trim($this->options['path']), "/");
+        $pathWithoutLastSegment = substr($pathWithoutLeadingAndTrailingSlashes, 0, strrpos($pathWithoutLeadingAndTrailingSlashes, "/"));
```

Comment:
Please use method `getPath()` instead of access `$this->options['path']` itself.